src/callbacks/callbacks.jl

```diff
@@ -122,13 +122,23 @@ function on(::BatchBegin, ::Phase, cb::ToGPU, learner)
 end
 
 
-garbagecollect() = (GC.gc(); ccall(:malloc_trim, Cvoid, (Cint,), 0))
+function garbagecollect()
+    GC.gc()
+    if Base.Sys.islinux()
+        ccall(:malloc_trim, Cvoid, (Cint,), 0)
+    end
+end
```

Comment:
If CUDA is in play, it is helpful to pair this with `CUDA.reclaim()`.

Reply:
This callback just exists to fix issues I had with memory leaks that weren't being cleaned by ´GC.gc()`, so I'd  leave that out for now. Or have you run into a situation where you had to periodically run  `CUDA.reclaim` during training?